storage/innobase/trx/trx0rseg.cc

```diff
@@ -240,6 +261,24 @@ bool trx_rseg_read_wsrep_checkpoint(XID& xid)
 
 	return found;
 }
+
+/** Clear WSREP XID in system header.
+For upgrading from versions prior to 10.3.5 */
+static void
+trx_rseg_clear_wsrep_sys_header_checkpoint()
+{
+	mtr_t mtr;
+	mtr.start();
+	const buf_block_t* sys = trx_sysf_get(&mtr);
+	ut_ad(mach_read_from_4(sys->frame + TRX_SYS + TRX_SYS_WSREP_XID_INFO
+			       + TRX_SYS_WSREP_XID_MAGIC_N_FLD)
+	      == TRX_SYS_WSREP_XID_MAGIC_N);
```

Comment:
If this needs to be a separate function, I’d suggest a shorter name, such as `trx_rseg_clear_wsrep_trx_sys`.
This clearing must be part of the same atomic mini-transaction that updates the undo log header pages. Thus, the `mtr_t::mtr_t()`, `mtr_t::start()` and `mtr_t::commit()` calls must be omitted.

What actually guarantees that the assertion will hold? Is it the call to `trx_rseg_init_wsrep_xid()`, from `trx_rseg_init_binlog_info()`?